mysql/resource_rds_config.go

```diff
@@ -0,0 +1,182 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceRDSConfig() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateRDSConfig,
+		UpdateContext: UpdateRDSConfig,
+		ReadContext:   ReadRDSConfig,
+		DeleteContext: DeleteRDSConfig,
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportRDSConfig,
+		},
+		Schema: map[string]*schema.Schema{
+			"binlog_retention_period": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     0,
+				Description: "Sets the number of hours to retain binary log files",
+			},
+			"replication_target_delay": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     0,
+				Description: "Sets the number of seconds to delay replication from source database instance to the read replica",
+			},
+		},
+	}
+}
+
+func CreateRDSConfig(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	db, err := getDatabaseFromMeta(ctx, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	for _, stmtSQL := range RDSConfigSQL(d) {
+		log.Println("Executing statement:", stmtSQL)
+
+		_, err = db.ExecContext(ctx, stmtSQL)
+		if err != nil {
+			return diag.Errorf("failed running SQL to set RDS Config: %v", err)
+		}
+	}
+
+	id := strconv.FormatInt(time.Now().UTC().UnixNano(), 10)
```

Comment:
Why are you using basically a "random" value? Maybe define a constant when it comes to it - not that it would mean anything, but if there is only one instance, it won't be wrong.

(This is only optional nit-picking; I believe the state should be the same when it contains exactly same resources - this breaks that assumption, but terraform would be fine with that)

Reply:
I thought that at some point one could use the same name and got into strange errors. A constant would be nice too. I will leave it like this for now.

Reply:
haha, ok. I will create a constant for that ;-)